### PR TITLE
Add support to enable read replicas on existing redis instances

### PR DESCRIPTION
**Pull request by himanikh**
Add ReadReplicasMode and secondaryIpRange support for Redis Update

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
redis:Added Multi read replica field `readReplicasMode` and `secondaryIpRange` in `google_redis_instance`
```

### CODE REVIEW

File: mmv1/products/redis/api.yaml [hunk 1]
```diff
@@ -285,3 +285,14 @@ objects:
         default_value: :READ_REPLICAS_DISABLED
         min_version: beta
         input: true
+      - !ruby/object:Api::Type::String
+        name: secondaryIpRange
+        description: |
+          Optional. Additional ip ranges for node placement, beyond those specified
```
Comment (melinath) [inline mmv1/products/redis/api.yaml:291]: These docs don't quite match what I see on https://cloud.google.com/memorystore/docs/redis/reference/rest/v1beta1/projects.locations.instances - just wanted to confirm that this is intentional? Generally we try to only deviate from c.g.c in terms of field names or if there's Terraform-specific information

File: mmv1/third_party/terraform/tests/resource_redis_instance_test.go [hunk 2]
```diff
@@ -240,6 +240,8 @@ resource "google_redis_instance" "test" {
     notify-keyspace-events = ""
   }
   redis_version = "REDIS_5_0"
+	read_replicas_mode = "READ_REPLICAS_ENABLED"
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_redis_instance_test.go:243]: The indentation here seems to use tabs - should be spaces.

**Review by melinath — CHANGES_REQUESTED** [review]
LGTM, except for a docs discrepancy and a potential issue with the proposed change to readReplicasMode.

**Comment by melinath** [comment]
@himanikh could you also rebase your PR off the latest master branch (or merge it in) - it looks like you're currently based off a version that has some issues in its CI build pipeline, so I can't run tests for this PR.

**Comment by melinath** [comment]
Sorry - removing the TF default would _not_ be a breaking change, since you'd be switching to a server-side default, which leaves whatever is in the state in the state.

File: mmv1/third_party/terraform/tests/resource_redis_instance_test.go (hunk 2 above)
Comment (himanikh) [reply mmv1/third_party/terraform/tests/resource_redis_instance_test.go:243]: fixed in https://github.com/GoogleCloudPlatform/magic-modules/pull/5793

File: mmv1/products/redis/api.yaml (hunk 1 above)
Comment (himanikh) [reply mmv1/products/redis/api.yaml:406]: fixed in https://github.com/GoogleCloudPlatform/magic-modules/pull/5793

**Comment by melinath** [comment]
superseded by https://github.com/GoogleCloudPlatform/magic-modules/pull/5793

**Pull request closed without merging**